### ISSUE TITLE
Cannot Login with Correct Password within Hash Function

### ISSUE THREAD

**Issue by JacobYim** [JacobYim/CSE442-BuyNSell#62]
Even though typed correct password with the email in login.html, it shows up the Login Failure messages.

**Comment by JacobYim** [comment]
It looks like the Hash Function issue. The code looks like the below.

```
app.post('/login.html', (req,res) => {
  var email = String(req.body['email']);
  var password = String(req.body['password']);
  if ((email != '' && email != ' ' && !email.includes(';') && !email.includes('=') && email.includes('@') && email.includes('.') && !email.includes("'" && !email.includes(';'))) && 
      (password != '' && password != ' ' && !password.includes(';') && !password.includes('.') && !password.includes('=') && !password.includes('(') && !password.includes(')')&& !password.includes("'"))){
        console.log(password);
      password = passwordHasher(password);
      console.log("Password is Secure......................."+password);
    
      db.query('SELECT * FROM user_profile where email=\''+email+'\' and password=\'' + password + '\'', function (err, rows, fields) {
      console.log(rows);
      if (!err) {
          // console.log(rows)
          if (rows.rowCount ==1 ) {
              res.send('Congrate!! Login Success!!');
              // res.redirect('/');
          } else {
              res.send('Login Failure');
              // res.redirect('/');
          }
      } else {
          res.send('error : ' + err);
          // res.redirect('/');
      }
    }); 
  }else{
    res.send('invalid input')
  }
});
```

**Comment by JacobYim** [comment]
```
password = passwordHasher(password);
```

It gives arbitrary hash, regardless the input is same.

**Comment by JacobYim** [comment]
To solve this USERCASE, it is discussed at the issue #63

**Comment by JacobYim** [comment]
This problem has been solved at Issue #63 and it has been merged at #61 Merge Request to `developer`
